File: API/API/repositorys/TaskAssignmentRepository.vb
Imports System.Data.Entity
Public Class TaskAssignmentRepository
    Implements ITaskAssignmentRepository

    Private ReadOnly _db As AppDbContext

    Public Sub New(db As AppDbContext)
        _db = db
    End Sub

    Public Function Exists(taskId As Integer, userId As Guid) As Boolean Implements ITaskAssignmentRepository.Exists

        Return _db.TaskAssignments.Any(
            Function(x) x.TaskId = taskId AndAlso
                        x.UserId = userId)

    End Function

    Public Function GetPermission(taskId As Integer, userId As Guid) As TaskAssignment Implements ITaskAssignmentRepository.GetPermission

        Return _db.TaskAssignments.FirstOrDefault(
            Function(x) x.TaskId = taskId AndAlso
                        x.UserId = userId)

    End Function

    Public Function GetAssignedTasks(userId As Guid) As List(Of TaskAssignment) Implements ITaskAssignmentRepository.GetAssignedTasks

        Return _db.TaskAssignments.
        Include(Function(x) x.Task).
        Where(Function(x) x.UserId = userId).
        OrderByDescending(Function(x) x.AssignedAt).
        ToList()

    End Function

    Public Sub Add(assign As TaskAssignment) Implements ITaskAssignmentRepository.Add

        _db.TaskAssignments.Add(assign)

    End Sub

    Public Sub Save() Implements ITaskAssignmentRepository.Save

        _db.SaveChanges()

    End Sub

End Class

File: API_v2/Repositorys/TaskAssignmentRepository.cs
using API_v2.Datas;
using API_v2.Models;
using API_v2.Repositorys.IRepositorys;
using Microsoft.EntityFrameworkCore;

namespace API_v2.Repositorys
{
    public class TaskAssignmentRepository : ITaskAssignmentRepository
    {
        private readonly AppDbContext _db;

        public TaskAssignmentRepository(AppDbContext db)
        {
            _db = db;
        }

        public bool Exists(int taskId, Guid userId)
        {
            return _db.TaskAssignments.Any(x => x.TaskId == taskId && x.UserId == userId);
        }

        public TaskAssignment? GetAssignment(int taskId, Guid userId)
        {
            return _db.TaskAssignments.FirstOrDefault(x => x.TaskId == taskId && x.UserId == userId);
        }

        public List<TaskAssignment> GetAssignedTasks(Guid userId)
        {
            return _db.TaskAssignments
                .Include(x => x.Task)
                .Where(x => x.UserId == userId)
                .OrderByDescending(x => x.AssignedAt)
                .ToList();
        }

        public void Add(TaskAssignment assignment)
        {
            _db.TaskAssignments.Add(assignment);
        }

        public void Remove(TaskAssignment assignment)
        {
            _db.TaskAssignments.Remove(assignment);
        }

        public void Save()
        {
            _db.SaveChanges();
        }
    }
}
